wasm/constants.py

```diff
@@ -1,39 +1,53 @@
-from .typing import (
-    SInt32,
-    SInt64,
-    UInt32,
-    UInt64,
-)
+import numpy
 
-UINT6_CEIL = UInt32(2 ** 6)
+UINT6_CEIL = numpy.uint32(2 ** 6)
```

Comment:
It might make sense to take advantage of numpy's built-in resources for getting these values.  For example,
```python
def get_int_range(dtype):
    iinfo = np.iinfo(dtype)
    return dtype(iinfo.min), dtype(iinfo.max)

def get_float_range(dtype):
    finfo = np.finfo(dtype)
    return finfo.min, finfo.max
```

Reply:
One caveat with this is that it's getting the value ranges that are dependent upon the hardware.  But I think that's essentially the case anyway since you're storing the values with numpy and numpy just stores things at the hardware level.

Reply:
Yeah, I had originally done it something similar to this but I converted to hard coded values so that we couldn't end up in a situation where a hardware difference caused the library to malfunction.